sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3509,20 +3531,15 @@ object Sequence {
       step: String,
       estimatedStep: String,
       len: String): String = {
-    val longLen = ctx.freshName("longLen")
+    val calcFn = "Sequence.sequenceLength"
```

Comment:
I'm surprised that this works. Should we do `classOf[Sequence].getName` + ".sequenceLength"?

Reply:
Not sure exactly how, but my hypothesis is that these factors all play a role:
* Sequence object ends up in the same compilation unit as this function (not sure if this is expected)
* `sequenceLength` function is effectively static in Scala and publicly accessible

I noticed some other functions in this file do this as well -- e.g. https://github.com/apache/spark/blob/128f5523194d5241c7b0f08b5be183288128ba16/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L613

I do like this better though, cleaner and easier to understand.